Remove metadata when dropping chunk

fabriziomello opened the pull request:
Historically we preserve chunk metadata because the old format of the
Continuous Aggregate has the `chunk_id` column in the materialization
hypertable so in order to don't have chunk ids left over there we just
mark it as dropped when dropping chunks.

In #4269 we introduced a new Continuous Aggregate format that don't
store the `chunk_id` in the materialization hypertable anymore so it's
safe to also remove the metadata when dropping chunk and all associated
Continuous Aggregates are in the new format.

Also added a post-update SQL script to cleanup unnecessary dropped chunk
metadata in our catalog.

Closes #6570

akuzm commented on test/sql/updates/post.catalog.sql:
```diff
@@ -45,20 +45,18 @@ SELECT count(*)
 FROM  _timescaledb_catalog.chunk c
 INNER JOIN pg_class cl ON (cl.oid=format('%I.%I', schema_name, table_name)::regclass)
-WHERE NOT c.dropped
+WHERE c.dropped IS FALSE
```
What's the difference between old and new, `c.dropped` can be null?

fabriziomello replied:
It cannot... just a cosmetic thing... sometimes I have some OCD regarding some coding styles... hehehehe... will preserve the original because it will not improve anything here and just add noise to repo history.

svenklemm reviewed (dismissed):
I dont think you should add this to post-update but use latest-dev instead to fix because otherwise it will constrain changes that can be done in the update script in the future

fabriziomello commented:
> I dont think you should add this to post-update but use latest-dev instead to fix because otherwise it will constrain changes that can be done in the update script in the future

Good that you mentioned it. I had the same feeling but ended up placing this here in order to make sure the future releases we'll don't have chunk metadata leftovers, but I can move it to the latest-dev and in the future when we complete remove the `dropped` column then write again this cleanup script to make sure we'll not end up with metadata leftovers.

fabriziomello commented:
@svenklemm moved the post-update.sql cleanup to latest-dev.sql

jnidzwetzki commented:
@fabriziomello Nit: there is a typo in the commit message `unecessary` -> `unnecessary`

jnidzwetzki commented on sql/updates/latest-dev.sql:
```diff
@@ -17,3 +17,52 @@ DROP FUNCTION IF EXISTS _timescaledb_functions.invalidation_process_hypertable_l
 DROP FUNCTION IF EXISTS _timescaledb_functions.hypertable_invalidation_log_delete(integer);
 
+-- Cleanup metadata for deleted chunks
+CREATE UNLOGGED TABLE _timescaledb_catalog._chunks_remove AS
```
Nit: Consider prefixing the table with `_tmp`

fabriziomello replied:
Turned it into a PL/pgSQL function!

jnidzwetzki commented:
@fabriziomello As discussed, I also did some experiments with compressed hypertables and CAggs. The clean-up works correctly as well.

I am wondering if this cleanup SQL should also be added to the cagg_migration steps and run every time you convert a CAgg into the finalized form.

fabriziomello commented:
> @fabriziomello As discussed, I also did some experiments with compressed hypertables and CAggs. The clean-up works correctly as well.
> 
> I am wondering if this cleanup SQL should also be added to the cagg_migration steps and run every time you convert a CAgg into the finalized form.

Very good catch @jnidzwetzki thanks. Will add the cleanup to the migration scripts.

fabriziomello commented:
> > @fabriziomello As discussed, I also did some experiments with compressed hypertables and CAggs. The clean-up works correctly as well.
> > I am wondering if this cleanup SQL should also be added to the cagg_migration steps and run every time you convert a CAgg into the finalized form.
> 
> Very good catch @jnidzwetzki thanks. Will add the cleanup to the migration scripts.

Added it also to the cagg_migrate procedure and also added proper regression tests
